File: app/src/main/java/com/litroenade/yunjiweather/data/repository/HomeWeatherSource.java
package com.litroenade.yunjiweather.data.repository;

import com.litroenade.yunjiweather.common.UiState;
import com.litroenade.yunjiweather.data.model.HomeWeatherData;

public interface HomeWeatherSource {
    UiState<HomeWeatherData> loadHomeWeather(String locationId, String cityName, double latitude, double longitude);
}

File: app/src/main/java/com/litroenade/yunjiweather/data/repository/HomeWeatherSource.kt
package com.litroenade.yunjiweather.data.repository

import com.litroenade.yunjiweather.common.UiState
import com.litroenade.yunjiweather.data.model.HomeWeatherData

interface HomeWeatherSource {
    fun loadHomeWeather(
        locationId: String,
        cityName: String,
        latitude: Double,
        longitude: Double
    ): UiState<HomeWeatherData>
}
